test/test_init_teardown.c

```diff
@@ -0,0 +1,436 @@
+    if (umfInit == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfTearDown", (void **)&umfTearDown);
+    if (umfTearDown == NULL) {
+        goto err_dlclose;
+    }
+
+    // initialize libumf (increment the reference counter of users)
+    (*umfInit)();
+
+    load_symbol(h_umf, "umfMemoryProviderCreateFromMemspace",
+                (void **)&umfMemoryProviderCreateFromMemspace);
+    if (umfMemoryProviderCreateFromMemspace == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfMemoryProviderDestroy",
+                (void **)&umfMemoryProviderDestroy);
+    if (umfMemoryProviderDestroy == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfPoolCreate", (void **)&umfPoolCreate);
+    if (umfPoolCreate == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfPoolDestroy", (void **)&umfPoolDestroy);
+    if (umfPoolDestroy == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfPoolMalloc", (void **)&umf_alloc);
+    if (umf_alloc == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfPoolFree", (void **)&umf_free);
+    if (umf_free == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfScalablePoolOps", (void **)&umfScalablePoolOps);
+    if (umfScalablePoolOps == NULL) {
+        goto err_dlclose;
+    }
+
+    load_symbol(h_umf, "umfMemspaceHostAllGet",
+                (void **)&umfMemspaceHostAllGet);
+    if (umfMemspaceHostAllGet == NULL) {
+        goto err_dlclose;
+    }
+
+    memspaceHostAll = (*umfMemspaceHostAllGet)();
+    if (memspaceHostAll == NULL) {
+        fprintf(stderr, "umf_load: memspaceHostAll NOT found\n");
+        goto err_dlclose;
+    }
+    fprintf(stderr, "umf_load: got memspace: memspaceHostAll\n");
+
+    ret = (*umfMemoryProviderCreateFromMemspace)(memspaceHostAll, NULL,
+                                                 &umf_provider_default);
+    if (ret || umf_provider_default == NULL) {
+        fprintf(stderr, "umf_load: error creating the default provider: %i\n",
+                ret);
+        goto err_dlclose;
+    }
+    fprintf(stderr, "umf_load: the default provider created from memspace\n");
+
+    umf_default = NULL;
+    ret = (*umfPoolCreate)((*umfScalablePoolOps)(), umf_provider_default, NULL,
+                           0, &umf_pool_default);
+    if (ret || umf_pool_default == NULL) {
+        fprintf(stderr, "umf_load: error creating the default pool: %i\n", ret);
+        goto err_destroy_provider;
+    }
+    fprintf(stderr, "umf_load: the default pool created from the "
+                    "All Nodes provider\n");
+
+    umf_default = umf_pool_default; // umf pool using the default memspace
+
+    // the highest capacity memspace
+    umf_large_cap = NULL;
+    load_symbol(h_umf, "umfMemspaceHighestCapacityGet",
+                (void **)&umfMemspaceHighestCapacityGet);
+    if (umfMemspaceHighestCapacityGet) {
```

Comment:
This symbol should be retrieved always, right? If so, then fail the test if it's NULL.

Reply:
Done.